src/lib/pubkey/xmss/xmss_parameters.h

```diff
@@ -26,18 +26,18 @@ class BOTAN_PUBLIC_API(2,0) XMSS_Parameters
    public:
       enum xmss_algorithm_t
          {
-         XMSS_SHA2_256_W16_H10 = 0x01000001,
-         XMSS_SHA2_256_W16_H16 = 0x02000002,
-         XMSS_SHA2_256_W16_H20 = 0x03000003,
-         XMSS_SHA2_512_W16_H10 = 0x04000004,
-         XMSS_SHA2_512_W16_H16 = 0x05000005,
-         XMSS_SHA2_512_W16_H20 = 0x06000006,
-         XMSS_SHAKE128_W16_H10 = 0x07000007,
-         XMSS_SHAKE128_W16_H16 = 0x08000008,
-         XMSS_SHAKE128_W16_H20 = 0x09000009,
-         XMSS_SHAKE256_W16_H10 = 0x0a00000a,
-         XMSS_SHAKE256_W16_H16 = 0x0b00000b,
-         XMSS_SHAKE256_W16_H20 = 0x0c00000c
+         XMSS_SHA2_10_256 = 0x00000001,
+         XMSS_SHA2_16_256 = 0x00000002,
```

Comment:
So the format changed between draft-16 and RFC? :/

Is there any reasonable way we can continue supporting older keys/sigs? It seems this change would invalidate all existing uses.

At the very least we'll also want to use a different OID for keys so it is not possible to mix them.

Reply:
Yes, unfortunately the OID format did change. There exists another, new draft status document of interest regarding XMSS [Algorithm Identifiers for HSS and XMSS for Use in the Internet X.509 Public Key Infrastructure](https://tools.ietf.org/html/draft-vangeest-x509-hash-sigs-03). Also somewhere down the road XMSS will likely be assigned an "official" OID. So this may not be the last incompatible change to XMSS we see. There is also a disclaimer in RFC 8391:

> This document is not an Internet Standards Track specification; it is
> published for informational purposes.

>   This document is a product of the Internet Research Task Force
>   (IRTF).  The IRTF publishes the results of Internet-related research
>   and development activities.  These results might not be suitable for
>   deployment.  This RFC represents the consensus of the Crypto Forum
>   Research Group of the Internet Research Task Force (IRTF).  Documents
>   approved for publication by the IRSG are not candidates for any level
>   of Internet Standard; see Section 2 of RFC 7841.

>   Information about the current status of this document, any errata,
>   and how to provide feedback on it may be obtained at
>   https://www.rfc-editor.org/info/rfc8391.

A possibility to support older formats would be to integrate a key conversion utility into the botan cli.